src/AddNode.js

```diff
@@ -12,61 +12,36 @@ type AddNodeProps = {
     disabled: boolean,
     'aria-label': string,
     isDraggingCommand: boolean,
+    closestAddNodeIndex: number,
     onClick: (stepNumber: number) => void,
     onDrop: (stepNumber: number) => void
 };
 
 const AddNode = React.forwardRef<AddNodeProps, HTMLDivElement>(
     (props, ref) => {
-        const [isDragOver, setIsDragOver] = useState(false);
-
-        /* istanbul ignore next */
-        const handleDragOver = (e: SyntheticDragEvent<HTMLButtonElement>) => {
-            if (props.isDraggingCommand) {
-                e.preventDefault();
-                setIsDragOver(true);
-            }
-        };
+        const [isDragOver] = useState(false);
 
         const handleClick = (e: SyntheticEvent<HTMLButtonElement>) => {
             const stepNumber = parseInt(e.currentTarget.dataset.stepnumber, 10);
             props.onClick(stepNumber);
         };
 
-        /* istanbul ignore next */
-        const handleDrop = (e: SyntheticDragEvent<HTMLButtonElement>) => {
-            e.preventDefault();
-            if (!props.disabled) {
-                const stepNumber = parseInt(e.currentTarget.dataset.stepnumber, 10);
-                props.onDrop(stepNumber);
-            }
-            setIsDragOver(false);
-        };
-
-        /* istanbul ignore next */
-        const handleDragLeave = (e: SyntheticDragEvent<HTMLButtonElement>) => {
-            e.preventDefault();
-            setIsDragOver(false);
-        };
-
         /* istanbul ignore next */
         const addNodeClasses = classNames(
             props.isDraggingCommand && 'AddNode--is-dragging-command'
         );
 
-        if (props.expandedMode || (isDragOver && !props.disabled)) {
+        const isNearestDragNode = ( props.programStepNumber === props.closestAddNodeIndex);
+        if (props.expandedMode || isNearestDragNode || (isDragOver && !props.disabled)) {
             return (
                 <div className={addNodeClasses}>
                     <div className='AddNode__drop-area-container'>
```

Comment:
We should be able to go further with the simplification of the `AddNode` component: now that it is no longer responsible for drag and drop, we can remove the `AddNode__drop-area-container` divs (this one and the one below for the collapsed case). And we can remove the associated style rules from `AddNode.scss`.

Reply:
Done.